File: app/src/main/java/com/adrianiglesia/atqr/ViewModel/LoginViewModel.java
package com.adrianiglesia.atqr.ViewModel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.adrianiglesia.atqr.Model.User;
import com.adrianiglesia.atqr.Services.MainRepository;

public class LoginViewModel extends ViewModel {

    private MutableLiveData<User> userMutableLiveData = new MutableLiveData<>();

    private MainRepository repository = new MainRepository();

    public void callLogin(Long document, String pass, ErrorCallback callback){
        repository.userLogin(document, pass, new MainRepository.LoginInterface() {
            @Override
            public void onSuccess(User user) {
                userMutableLiveData.setValue(user);
            }

            @Override
            public void onError(String msg) {
                callback.onErrorLogin(msg);

            }
        });
    }


    public LiveData<User> getUserLiveData(){
        return userMutableLiveData;
    }

    public interface ErrorCallback{
        void onErrorLogin(String msg);
    }
}

File: app/src/main/java/com/adrianiglesia/atqr/viewmodel/LoginViewModel.kt
package com.adrianiglesia.atqr.viewmodel

import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.adrianiglesia.atqr.model.User

import com.adrianiglesia.atqr.services.MainRepository

class LoginViewModel : ViewModel() {

    private val repository = MainRepository()

    var isLoading:MutableLiveData<Boolean> = MutableLiveData()
    var userMutableLiveData:MutableLiveData<User> = MutableLiveData()
    var message:MutableLiveData<String> = MutableLiveData()


    fun login(document: Long, pass: String){
        when {
            validate(document,pass) -> callLogin(document, pass)
            else -> message.value = "Verifique los campos ingresados"
        }
    }


    private fun callLogin(document: Long, pass: String) {
        isLoading.value = true

        repository.userLogin(document, pass, {
            message.value = "Exito!"
            userMutableLiveData.value = it
            isLoading.value = false


        }, {

            message.value = it
            isLoading.value = false

        } )
    }

    private fun validate(doc:Long, pass:String):Boolean{
        var valid = true

        if (doc == null || doc.toString().length < 8 ) {
            valid = false
        }

        if (pass.isEmpty() || pass.length < 8 || pass.length > 16) {
            valid = false
        }

        return valid
    }

}
